### PR TITLE
Make migrations between completely different enums with same name work

### PR DESCRIPTION
**Pull request by tomhoule**
closes https://github.com/prisma/prisma/issues/6484

### CODE REVIEW

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs [hunk 1]
```diff
@@ -549,6 +534,47 @@ impl<'schema> SqlSchemaDiffer<'schema> {
...
+                previous_usages_as_default.push((
+                    (columns.previous.table().table_index(), columns.previous.column_index()),
+                    next_usage_as_default,
+                ));
+            }
+        }
+
+        alter_enum.previous_usages_as_default = previous_usages_as_default;
+    }
+}
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:576]: do we want this kind of code? I think we could do better, thinking about how.

Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:576]: You mean mutating things in a weird order? I also find it a bit stressful to follow...

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:576]: `I think the order is very straightforward (we just accumulate in one vector), it's the repetition that I find distressing.`

Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:576]: Yeah, I can't figure out how much repetition there is already, but this looks like an important fix to get to the release. Do you want to hurry it, or do another round of refactoring still?

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:576]: Nah I can't get it better in this PR, it has to be more systematic — I'll explore that in the learning day.

**Pull request merged**